recipes/openvino/all/conanfile.py

```diff
@@ -0,0 +1,402 @@
+
+    @property
+    def _preprocessing_available(self):
+        return Version(self.version) < "2023.2.0"
+
+    @property
+    def _onnx_version(self):
+        if Version(self.version) == "2023.1.0":
+            return "1.13.1"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "9",
+            "apple-clang": "11",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version]["openvino"], strip_root=True)
+        get(self, **self.conan_data["sources"][self.version]["onednn_cpu"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/onednn")
+        get(self, **self.conan_data["sources"][self.version]["mlas"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/mlas")
+        get(self, **self.conan_data["sources"][self.version]["arm_compute"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/ComputeLibrary")
+        get(self, **self.conan_data["sources"][self.version]["onednn_gpu"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_gpu/thirdparty/onednn_gpu")
+        apply_conandata_patches(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if not self._gpu_option_available:
+            del self.options.enable_gpu
+
+    def configure(self):
+        suffix = "" if Version(conan_version).major < "2" else "/*"
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if self._protobuf_required:
+            if self.options.shared:
+                # we need to use static protobuf to overcome potential issues with multiple registrations inside
+                # protobuf when frontends (implemented as plugins) are loaded multiple times in runtime
+                self.options[f"protobuf{suffix}"].shared = False
+        # disable GPU plugin when clang is used; plugin has issues with static variables initialization
+        if self.settings.compiler == "clang" and self.options.get_safe("enable_gpu"):
+            self.options.enable_gpu = False
+
+    def build_requirements(self):
+        if self._target_arm:
+            self.tool_requires("scons/[>=4.2.0]")
+        if cross_building(self):
+            if self._protobuf_required:
+                self.tool_requires("protobuf/<host_version>")
+            if self.options.enable_tf_lite_frontend:
+                self.tool_requires("flatbuffers/<host_version>")
+        if not self.options.shared:
+            self.tool_requires("cmake/[>=3.18]")
+
+    def requirements(self):
+        self.requires("onetbb/[>=2021.3.0]")
```

Comment:
version ranges are still new and there are some extra rules that exist :) https://github.com/conan-io/conan-center-index/blob/46c64a6ce06c8bc97ffdff8a96ef694069103fc4/docs/adding_packages/dependencies.md?plain=1#L176

Reply:
is it allowed to use version ranges at least for tools?

Reply:
fixed